src/components/Form/FormProvider.tsx

```diff
@@ -207,7 +213,10 @@ function FormProvider(
             return;
         }
 
+        isSubmittingRef.current = true;
         KeyboardUtils.dismiss().then(() => onSubmit(trimmedStringValues));
+
+        return () => (isSubmittingRef.current = false);
```

Comment:
Well, wait, this callback will never be executed, because it's not a `useEffect` (it's `useCallback` and `useCallback` doesn't fire internal callbacks). Correct me if I'm wrong.

Reply:
In that case only option left here is to add a small delay before changing value

setTimeout(() => (isSubmittingRef.current = false),100);

Reply:
@ChavdaSachin I think a slightly better approach would be to use `useDebounce` hook with interval 500ms? What do you think about it?

Reply:
@kirillzyusko in that case I tried various methods , even async - await mechanism and KeyboardUtils.dismiss().then(() => onSubmit(trimmedStringValues)).finally(isSubmittingRef.current=false); but problem stays,
The only option worked for me was not resetting the value(note. all the onSubmit functions are associated with some sort of navigation, so it's working, and useSingleExecution hook also).

If that sounds an unacceptable solution then please take over the issue, sorry for the inconvenience in that case.

Reply:
Thanks @ChavdaSachin for your attempts to solve this problem, appreciate it!

In the end we decided to go in a slightly different way: https://github.com/Expensify/App/pull/53227